### PR TITLE
feat: Admin Shipping Profiles API

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/medusa/src/api-v2/admin/shipping-profiles/[id]/route.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse<AdminShippingProfileResponse>
+) => {
+  const fulfillmentService = req.scope.resolve<IFulfillmentModuleService>(
+    ModuleRegistrationName.FULFILLMENT
+  )
+
+  const shippingProfile = await fulfillmentService.retrieveShippingProfile(
+    req.params.id
+  )
```
Comment (pKorsholm) [inline packages/medusa/src/api-v2/admin/shipping-profiles/[id]/route.ts]: question(non-blocking): did we move away from using `remoteQuery` for the retrieve endpoints?

File: packages/medusa/src/api-v2/admin/shipping-profiles/middlewares.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import {
+  AdminCreateShippingProfile,
+  AdminShippingProfileParams,
+  AdminShippingProfilesParams,
+} from "./validators"
+
+export const adminShippingProfilesMiddlewares: MiddlewareRoute[] = [
+  {
+    matcher: "/admin/shipping-profiles*",
+    middlewares: [authenticate("admin", ["bearer", "session"])],
```
Comment (pKorsholm) [inline packages/medusa/src/api-v2/admin/shipping-profiles/middlewares.ts]:
```suggestion
    middlewares: [authenticate("admin", ["bearer", "session", "api-key"])],
```

File: packages/medusa/src/api-v2/admin/shipping-profiles/route.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+) => {
+  const fulfillmentService = req.scope.resolve<IFulfillmentModuleService>(
+    ModuleRegistrationName.FULFILLMENT
+  )
+
+  const [shippingProfiles, count] =
+    await fulfillmentService.listAndCountShippingProfiles(
+      req.filterableFields,
+      req.listConfig
+    )
```
Comment (pKorsholm) [inline packages/medusa/src/api-v2/admin/shipping-profiles/route.ts]: same comment about remoteQuery

**Review by pKorsholm — APPROVED** [review]
lgtm, nothing blocking 😄

**Review by fPolic — APPROVED** [review]

File: packages/medusa/src/api-v2/admin/shipping-profiles/[id]/route.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa/src/api-v2/admin/shipping-profiles/[id]/route.ts]: Updated, nice catch!

**Pull request merged**